querybook/webapp/redux/comment/reducer.ts

```diff
@@ -0,0 +1,97 @@
+import { produce } from 'immer';
+
+import {
+    commentStateKeyByEntityType,
+    IComment,
+    ICommentRaw,
+} from 'const/comment';
+import { convertRawToContentState } from 'lib/richtext/serialize';
+
+import { CommentAction, ICommentState } from './types';
+
+const initialState: ICommentState = {
+    cellIdToCommentIds: {},
+    tableIdToCommentIds: {},
+    commentsById: {},
+};
+
+function commentReducer(state = initialState, action: CommentAction) {
+    return produce(state, (draft) => {
+        switch (action.type) {
+            case '@@comment/RECEIVE_COMMENTS': {
+                const { comments } = action.payload;
+
+                const receivedComments: Record<number, IComment> = {};
+                comments.forEach(
+                    (comment: ICommentRaw) =>
+                        (receivedComments[comment.id] = {
+                            ...comment,
+                            text: convertRawToContentState(comment.text),
+                        })
+                );
+
+                draft.commentsById = {
```

Comment:
you don't need to spread:
```ts
for (const comment of comments) {
    draft.commentsById[comment.id] = {
        ...comment,
        text: convertRawToContentState(comment.text),
    }
}

```

Reply:
is there a reason loop is preferred over forEach?